users/views.py

```diff
@@ -1,3 +1,72 @@
-from django.shortcuts import render
+import json
+import bcrypt
+import jwt
 
-# Create your views here.
+from django.http              import JsonResponse
+from django.views             import View
+from core.utils               import *
+
+from django.core.exceptions   import ValidationError 
+from wesop.settings           import SECRET_KEY,ALGORITHM
+from core.validator import *
+
+from users.models import *
+
+
+class SignupView(View): 
+    def post(self,request):
+        
+        try:
+            
+            data       = json.loads(request.body)   
+            email      = data['email'] 
+            first_name = data['first_name'] 
+            last_name  = data['last_name']
+            password   = data['password']
+            
+            
+            if User.objects.filter(email = email).exists():
+             return JsonResponse({"message":"DUPLICATE_EMAIL"}, status = 400)
+            
+            validate_email(email)
+            validate_name(first_name)
+            validate_name(last_name)
+            validate_password(password)
+            
+            User.objects.create(
+                    skin_type    = Skintype.objects.get(skin_type=data['skin_type']),
+                    first_name   = first_name,
+                    last_name    = last_name,
+                    email        = email,
+                    password     = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8'),
+            )
```

Comment:
- 스킨 타입은 이따가 구두로 조금 더 자세히 이야기 하시죠
- 줄 바꿈 이후에 indent는 4칸으로 해주시면 됩니다.
```suggestion
            User.objects.create(
                skin_type    = Skintype.objects.get(skin_type=data['skin_type']),
                first_name   = first_name,
                last_name    = last_name,
                email        = email,
                password     = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8'),
            )
```